### PR TITLE
add AMP from pt 1.6

### PR DESCRIPTION
**Pull request by Borda**
add amp training...

### CODE REVIEW

**Comment by Borda** [comment]
replace the loss did not help...
https://discuss.pytorch.org/t/understanding-cdist-function/76296/15
